### ISSUE TITLE
Spring-web compatibility issues after upgrading from 2.1.3 to 2.1.7

### ISSUE THREAD

**Issue by 1988andy** [spring-projects/spring-framework#23592]
When I upgraded the spring-boot version from 2.1.3 to 2.1.7, I found that the `spring-web` version was upgraded from 5.1.5 to 5.1.9. However, when I deployed the project to the server and started tomcat, an error occurred:

```
org.xml.sax.SAXParseException; systemId: jar:file:/opt/apache-tomcat-8.5.33_1/webapps/.../WEB-INF/lib/spring-web-5.1.9.RELEASE.jar!/META-INF/web-fragment.xml; lineNumber: 5; columnNumber: 41; schema_reference.4: Failed to read schema document 'https://java.sun.com/xml/ns/javaee/web-fragment_3_0.xsd', because 
1) could not find the document; 
2) the document could not be read; 
3) the root element of the document is not <xsd:schema>.
at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.warning(ErrorHandlerWrapper.java:99)
at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:392)
at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:327)
at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:284)
at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.findSchemaGrammar(XMLSchemaValidator.java:2451)
at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.handleStartElement(XMLSchemaValidator.java:1768)
at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaValidator.startElement(XMLSchemaValidator.java:741)
at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:374)
at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:613)
at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:31
```

The pom.xml file starts like this:

```xml
<? The XML version = "1.0" encoding = "utf-8"? >
The < project XMLNS = "http://maven.apache.org/POM/4.0.0" XMLNS: xsi = "http://www.w3.org/2001/XMLSchema-instance"
Xsi: schemaLocation = "http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" >
```

What is the reason for this, or how can I change it to upgrade correctly?

**Comment by szegad** [comment]
We have similar issue upgrading from Spring 5.1.4 to 5.2.1
Tomcat 9 doesn't startup properly.
`WARNING [main] org.apache.tomcat.util.descriptor.XmlErrorHandler.logFindings Warning [schema_reference.4: Failed to read schema document 'https://java.sun.com/xml/ns/javaee/web-fragment_3_0.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.] reported processing [jar:file:/opt/tc/webapps/osp/WEB-INF/lib/spring-web-5.2.1.RELEASE.jar!/META-INF/web-fragment.xml].`

web-fragment.xml:
`<?xml version="1.0" encoding="ISO-8859-1"?>
<web-fragment xmlns="http://java.sun.com/xml/ns/javaee"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://java.sun.com/xml/ns/javaee https://java.sun.com/xml/ns/javaee/web-fragment_3_0.xsd"
	version="3.0" metadata-complete="true">
	<name>spring_web</name>
	<distributable/>
</web-fragment>`
where **https**://java.sun.com/xml/ns/javaee/web-fragment_3_0.xsd is the offending change.

**Comment by szegad** [comment]
`-Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=false`
is a workaround although we do not want to keep it like that for long.
Any ideas will be appreciated.

**Comment by rstoyanchev** [comment]
This appears to have required a fix on the Tomcat side in 9.0.18+, see https://bz.apache.org/bugzilla/show_bug.cgi?id=63311.
